content/post/the-future-is-rusty.en.md

```diff
@@ -0,0 +1,69 @@
+Rust is a language that guarantees to deny memory leaks and data races by compile-time checks without a Garbage Collector, using the concept of Ownership & Borrowing.
+Rust also has zero-cost abstraction which allows for almost C-like performance.
+
+Call me a fanboy of Rust but here are some stats for the analytical minds to gnaw at, 
+[StackOverFlow Survey for Most Loved Language:](https://insights.stackoverflow.com/survey/2018#most-loved-dreaded-and-wanted) And... the victor is, you guessed it, Rust.
+While C and C++ score really high on the list of most dreaded languages.
+Also if you check you will find that the stats are quite the same for the few years since Rust 1.0 came out and Rust became a stable language.
+
+#### But I heard Go is backed by Google. And it can also give C-like performance. So how come you aren't selling Go?
+Because Rust has the backing of Mozilla group(the people behind Firefox). And Go is extremely opinionated for it to be viable for System Level Programming(in my view). This means that Go is better used for higher level tasks. Also considering it has a garbage collector attached to its hip.
+
+![put-your-trust-in-rust](https://blog.mozilla.org/firefox/files/2017/03/Put-Your-Trust-in-Rust.png)
+
+### So how does Rust provide memory safety?
+It's elementary my dear Watson.
+Rust has compile-time reference counting and "Ownership and Borrowing". Which allows it to check the code for possible mistakes, and clear up memory without the need for clunky garbage collectors.
+And yes, it's quite similar to what you refer to as unique_pointer in C++. But the difference lies in the fact that in Rust it's a compile-time check whereas in C++ it's performed at runtime.
+
+This comes from the observation that memory becomes `unsafe` when we have aliasing(pointer/reference) of data which is then mutated and then very likely leaves us behind with dangling hope... *cough* I mean pointer.
+![guaranteeing-memory-safety-in-rust](https://image.slidesharecdn.com/rustoverview-140913055618-phpapp02/95/guaranteeing-memory-safety-in-rust-13-638.jpg?cb=1410587857)
+
+So now we have some idea how taking away shared mutability and using the Ownership and Borrowing model helps us with memory safety in Rust.
+
+
+### But I like my language. Why should I switch to Rust?
+So at this point, I am sure you think that Rust is just the same old candy in a new wrapper. But that's pretty ill-advised, Rust has a bunch of amazing features that make it worth the switch.
+One such thing is the lovely Web Assembly. Which has become the way to go for the performance intensive frontend web tasks. And Rust has the best integration possible with the format which makes it the clear choice for the performance hungry web developers.
+
+
+So is that all? Not at all, Rust is a rather fascinating language, and I will surely be covering it in much greater detail in the coming months. Ciao!

```

Comment:
I would like to know more, throw me some links to learn more.

Reply:
I am doing a post about that too. Will forward you the draft.